kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -156,16 +193,21 @@ private RegisteredSchema getSchemaFromRawSchema(
       boolean isKey) {
     checkArgument(format.requiresSchema(), "%s does not support schemas.", format);
 
-    ParsedSchema schema =
-        format
-            .getSchemaProvider()
-            .parseSchema(rawSchema, /* references= */ emptyList(), /* isNew= */ true)
-            .orElseThrow(
-                () ->
-                    Errors.invalidSchemaException(
-                        String.format(
-                            "Error when parsing raw schema. format = %s, schema = %s",
-                            format, rawSchema)));
+    ParsedSchema schema;
+    try {
+      schema =
+          format
+              .getSchemaProvider() // throws UnsupportedOperationException for bad config
```

Comment:
Same thing about minimizing scope of try-block.

Reply:
As above